src/core/lib/security/authorization/matchers.cc

```diff
@@ -130,6 +130,23 @@ bool NotAuthorizationMatcher::Matches(const EvaluateArgs& args) const {
 }
 
 bool HeaderAuthorizationMatcher::Matches(const EvaluateArgs& args) const {
+  if (matcher_.name() == HttpMethodMetadata().key()) {
```

Comment:
Thank you @yashykt for making the updates. I thought GetHeaderValue would be able to fetch :method, :authority and :path too. But recently, when I was working on the PR related to A41 proposal, I realized there is some difference. I was just curious, why the change in the handling of certain metadata keys?

Also I think instead of adding these changes here, we could update EvaluateArgs::GetHeaderValue
https://github.com/grpc/grpc/blob/master/src/core/lib/security/authorization/evaluate_args.cc#L117
```
if (key == HttpMethodMetadata().key()) {
  return GetMethod();
}
// similarly for :path and :authority
```
And we don't have to change HeaderAuthorizationMatcher::Matches.

Reply:
Done. This seems to be a temporary requirement, so I've added a TODO to remove the special case after #28267 is merged. The tests seem like a nice addition anyway.